lib/main.dart

```diff
@@ -1,67 +1,305 @@
-
+import 'package:flutter/services.dart';
 void main() {
-  runApp(const MyApp());
+  WidgetsFlutterBinding.ensureInitialized();
+  SystemChrome.setPreferredOrientations([DeviceOrientation.landscapeLeft])
+      .then((_) {
+  runApp(MyApp());
+  });
 }
 
+var marginDivider = 1;
+double orientationWidth = 1;
+double orientationHeight = 1;
+
 class MyApp extends StatelessWidget {
   const MyApp({Key? key}) : super(key: key);
-
-  // This widget is the root of your application.
   @override
   Widget build(BuildContext context) {
     return MaterialApp(
       title: 'Flutter Demo',
       theme: ThemeData(
         primarySwatch: Colors.blue,
       ),
-      home: const MyHomePage(title: 'Flutter Demo Home Page'),
+      home: const HomePage(),
     );
   }
 }
 
-class MyHomePage extends StatefulWidget {
-  const MyHomePage({Key? key, required this.title}) : super(key: key);
-  final String title;
+class HomePage extends StatefulWidget {
+  const HomePage({Key? key}) : super(key: key);
 
   @override
-  State<MyHomePage> createState() => _MyHomePageState();
+  _HomePageState createState() => _HomePageState();
 }
 
-class _MyHomePageState extends State<MyHomePage> {
-  int _counter = 0;
+Widget _imageContainerStart(BuildContext context, String link, String gameName,
+    String gameShortDescription) {
+  var phoneSize = MediaQuery.of(context).size;
+  if (MediaQuery.of(context).orientation == Orientation.landscape) {
+    orientationWidth = 0.9;
+    orientationHeight = 1.24;
+    marginDivider = 2;
+  }
+  return Container(
+      width: phoneSize.width * 0.83 * orientationWidth,
+      height: phoneSize.height * 0.45 * orientationHeight,
+      margin: const EdgeInsets.all(16.0),
+      decoration: BoxDecoration(
+          borderRadius: const BorderRadius.all(Radius.circular(5.5)),
+          image: DecorationImage(image: AssetImage(link), fit: BoxFit.cover)),
+      child: Container(
+        margin: EdgeInsets.only(
+            left: 12.0, right: 17.0, top: 140.0 / marginDivider),
+        child: Column(
+          children: [
+            Align(
+                alignment: Alignment.centerLeft,
+                child: Text(
+                  gameName,
+                  style:
+                      const TextStyle(fontSize: 42, color: Color(0xFFf2f2f2)),
+                )),
+            Align(
+                alignment: Alignment.centerLeft,
+                child: Text(
+                  gameShortDescription,
+                  style:
+                      const TextStyle(fontSize: 20, color: Color(0xFFf2f2f2)),
+                )),
+          ],
+        ),
+      ));
+}
 
-  void _incrementCounter() {
-    setState(() {
-      _counter++;
-    });
+Widget _imageContainerSecond(BuildContext context, String link, String gameName,
+    String gameShortDescription) {
+  var phoneSize = MediaQuery.of(context).size;
+  if (gameName.length > 9) {
+    gameName = gameName.substring(0, 9) + "...";
+  }
```

Comment:
it’s not a bad solution, but usually, in UI design, we don’t check the length of the string so that you can use this for reference: [link](https://stackoverflow.com/questions/53359109/dart-how-to-truncate-string-and-add-ellipsis-after-character-number)

Reply:
You shoudn't handle ellipsis manually (in imperative way), because this problem is already solved for you, and you can use declarative way (`new Text(overflow: TextOverflow.ellipsis)`). More details [here](https://stackoverflow.com/questions/44579918/flutter-wrap-text-on-overflow-like-insert-ellipsis-or-fade)